Do not add -latomic ever; macOS does not have it, not even on ARM.
Reverts https://github.com/swoole/swoole-src/commit/741ace2990c5f22f098af9f874bebbdf74a342a5
Only affects PHP 7.3.x and earlier.
https://github.com/openswoole/swoole-src/issues/131
--- config.m4.orig	2023-03-02 07:00:41.000000000 -0600
+++ config.m4	2023-03-03 23:13:10.000000000 -0600
@@ -809,10 +809,6 @@
         CXXFLAGS="$CXXFLAGS -std=c++11"
     fi
 
-    if test "$SW_CPU" = "arm"; then
-        PHP_ADD_LIBRARY(atomic, 1, SWOOLE_SHARED_LIBADD)
-    fi
-
     PHP_ADD_BUILD_DIR($ext_builddir/ext-src)
     PHP_ADD_BUILD_DIR($ext_builddir/src/core)
     PHP_ADD_BUILD_DIR($ext_builddir/src/memory)
